src/app/clusters/zone-management-server/zone-management-server.h

```diff
@@ -0,0 +1,377 @@
+     *    @brief Command Delegate for creation of a ZoneTrigger.
+     *
+     *   @param[in]  zoneTrigger    Structure with parameters for defining a ZoneTriggerControl.
+     *
+     *   @return Success if the creation is successful; otherwise, the command SHALL be
+     *   rejected with an appropriate error.
+     */
+    virtual Protocols::InteractionModel::Status CreateTrigger(const ZoneTriggerControlStruct & zoneTrigger) = 0;
+
+    /**
+     *    @brief Command Delegate for update of a ZoneTrigger.
+     *
+     *   @param[in]  zoneTrigger    Structure with parameters for defining a ZoneTriggerControl.
+     *
+     *   @return Success if the update is successful; otherwise, the command SHALL be
+     *   rejected with an appropriate error.
+     */
+    virtual Protocols::InteractionModel::Status UpdateTrigger(const ZoneTriggerControlStruct & zoneTrigger) = 0;
+
+    /**
+     *    @brief Command Delegate for the removal of a ZoneTrigger for a given zoneID.
+     *
+     *   @param[in] zoneID  Indicates the ID of the zone to remove the ZoneTrigger for.
+     *
+     *   @return Success if the removal is successful; otherwise, the command SHALL be
+     *   rejected with an appropriate error.
+     */
+    virtual Protocols::InteractionModel::Status RemoveTrigger(uint16_t zoneID) = 0;
+
+    /**
+     *   @brief Delegate callback for notifying change in an attribute.
+     *
+     */
+    virtual void OnAttributeChanged(AttributeId attributeId) = 0;
+
+    /**
+     *  @brief Callback into the delegate once persistent attributes managed by
+     *  the Cluster have been loaded from Storage.
+     */
+    virtual CHIP_ERROR PersistentAttributesLoadedCallback() = 0;
+
+    /**
+     * Delegate function to load the created zones and triggers.
+     * The application is responsible for persisting them. The Load APIs
+     * would be used to load the persisted zones and triggers into the cluster
+     * server list members at initialization.
+     * Once loaded, the cluster server can serve Reads on these
+     * attributes.
+     */
+    virtual CHIP_ERROR LoadZones(std::vector<ZoneInformationStorage> & aZones) = 0;
+
+    virtual CHIP_ERROR LoadTriggers(std::vector<ZoneTriggerControlStruct> & aTriggers) = 0;
+
+private:
+    friend class ZoneMgmtServer;
+
+    ZoneMgmtServer * mZoneMgmtServer = nullptr;
+
+    /**
+     * This method is used by the SDK to ensure the delegate points to the server instance it's associated with.
+     * When a server instance is created or destroyed, this method will be called to set and clear, respectively,
+     * the pointer to the server instance.
+     *
+     * @param aZoneMgmtServer  A pointer to the ZoneMgmtServer object related to this delegate object.
+     */
+    void SetZoneMgmtServer(ZoneMgmtServer * aZoneMgmtServer) { mZoneMgmtServer = aZoneMgmtServer; }
+
+protected:
+    ZoneMgmtServer * GetZoneMgmtServer() const { return mZoneMgmtServer; }
+};
+
+enum class OptionalAttribute : uint32_t
+{
```

Comment:
Enum OptionalAttribute is Empty, but you already use the version from zap in you code, 

CameraAvStreamManagement::OptionalAttribute::kImageFlipVertical)

why we still need to define our own version?

Reply:
Yes, it is not needed for this cluster as there is no optional attribute. Had created the skeleton initially. Removed it.